frontend/src/static/js/v_summary.js

```diff
@@ -348,7 +357,10 @@ window.vSummary = {
       // array of array, sorted by repo
       const full = [];
 
-      this.repos.forEach((repo) => {
+      // create deep clone of this.repos if merging groups
+      const groups = this.isMergeGroup ? JSON.parse(JSON.stringify(this.repos)) : this.repos;
```

Comment:
Without deep clone, merging groups will modify the original `repos`, which will trigger The Vue watcher `repos()`.https://github.com/reposense/RepoSense/blob/9b0f4f6600cf49543817d8803cf73dadd600856f/frontend/src/static/js/v_summary.js#L123-L125
Within the Vue watcher `repos()`, it will execute `this.getFiltered()` which will cause an infinite loop.

Reply:
> it is easier to merge the groups.

OK, makes sense.

Reply:
> Within the Vue watcher `repos()`, it will execute `this.getFiltered()` which will cause an infinite loop.

Do we really need the watcher for `repos()` now?

Reply:
Actually I'm also not sure. Depends on the author's purpose on placing a watcher for `repos()`.

Reply:
A watcher of large data like `repos()` will have performance drawbacks. Can you help investigate whether this is truly necessary? From my suspicion, we are already watching every flag which may affect repo(). And we do not have other methods which may directly modify `repos()`?

Reply:
Okay will look deeper into it.

But I have tried running the report without the `repos()` watcher. One thing to note is that opening the CodeView without the watcher will lead to `tabAuthor` and `tabRepo` hashes to not be in the url.
<img width="1667" alt="Screen Shot 2019-06-28 at 4 54 47 PM" src="https://user-images.githubusercontent.com/32864116/60330868-f8020600-99c5-11e9-8e2b-4544090bb1da.png">

Reply:
> Can you help investigate whether this is truly necessary?

**Current:**
The watcher `repos()` will be called when opening/closing tabs (authorship and zoom). This is because the opening and closing of tabs will lead to calling `getUsers()` method and set the `repos` as shown below. 
https://github.com/reposense/RepoSense/blob/369186b9b299ba6a41735b6d76d91391c0c4e92a/frontend/src/index.pug#L29-L35

The calling of `getUsers()` when opening/closing tabs is redundant as they will lead to the same contents of `repos` being set.

**Suggestion:** 
To remove the `repos()` watcher without affecting the current implementation, we can have a `users` data within the `app` component and calls `getUsers()` to set the `users` when the `app` component is initially created. Then we pass `users` instead of `getUsers()` [here](https://github.com/reposense/RepoSense/blob/369186b9b299ba6a41735b6d76d91391c0c4e92a/frontend/src/index.pug#L31) (to prevent `getUsers` to be executed repeatedly when opening/closing tabs).

Furthermore, I realised that there is some redundant code here in line [#117](https://github.com/reposense/RepoSense/blob/369186b9b299ba6a41735b6d76d91391c0c4e92a/frontend/src/static/js/main.js#L117) and [#131](https://github.com/reposense/RepoSense/blob/369186b9b299ba6a41735b6d76d91391c0c4e92a/frontend/src/static/js/main.js#L131) in main.js. It is setting `this.users` to be an empty array. But then `this.users` is not used and it does not exist in the data of the `app` component. Is it because previously there is a `users` data in the `app` component?

Reply:
I think the suggestion is acceptable.  You may start on #780 first before this PR.

@ongspxm You input?